### DISCUSSION TITLE
Help please: Restrict formats for alsa output - how to use allowed_formats

### DISCUSSION THREAD

**Discussion by 111MilesToGo** [MusicPlayerDaemon/MPD#1383] [Q&A] [answered]
Can someone please help? Unfortunately, I don't get to grips with the mpd documentation.

I am using a new USB-DAC, a Cayin RU6 dongle. I encounter some problems when going from one track to another one IF the bit depth changes. It is a hit-or-miss affair, some bit depth changes work, some fail producing just noise.

I am using the alsa output. I find that the problem can be solved for PCM files by enforcing 24-bit output. i.e. put a line
`format "*:24:*"` or `format "*24:2"` into my mpd.conf in the alsa output section.

Unfortunately, this line kills the ability to play DSD files correctly. The RU6 DAC natively plays DSD64-256, i.e. it accepts DSD64 and DSD128 as DoP. (There is no Linux suuport for native DSD with it.) Without this `format` line, DSD as DoP works fine.

So I think I should have a line with `allowed_formats` in my alsa output section. So sorry, I can't figure out how it should read. It shall account for `*:24:*` or `*:24:2` in case of PCM, as well as for DSD64 as DoP and DSD128 as DoP. Whatever I try like
`allowed_formats "*:24:* dsd64:=dop dsd128:=dop"`
lets mpd fail and Cantata throw a "it looks like your mpd isn't configured correctly".

Help, please, with how to write this `allowed_formats` line correctly.

**Comment by MaxKellermann** [comment]
"mpd fail" is not enough information to be able to help you.
See https://www.musicpd.org/help/

**Comment by 111MilesToGo** [comment, marked as answer]
Okay, sorry for not adhering to the rules. Shame on me.

But thank you, this triggered me to look at the verbose log of mpd. It said "failed to parse number of channels". Thus, the correct allowed_formats line in my alsa output should read:
`allowed_formats "*:24:* dsd64:*=dop dsd128:*=dop"`
in order to enforce 24-bit output for any PCM and play DSD as DoP as is possible given the rate restrictions of my Cayin RU6 DAC. With this line, everything works just fine.

Please note the tiny `*` following my `dsd64` and `dsd128` statements.

Dear Max, thus the mpd documentation needs a little correction. In the "Plugin reference" > "Output plugins" > "alsa" list of settings, the Example for the 
`allowed_formats F1 F2 …`
setting reads
`Example: “96000:16:* 192000:24:* dsd64:=dop *:dsd:”`.
However, the first dsd example should read
`dsd64:*=dop` - note the `*`.
I did not check the second dsd example
`*:dsd:”`
for correctness, although it looks suspicious, too.
